### PR TITLE
Feature/decorators simplify examples

### PR DESCRIPTION
**Pull request by jlnav**
Tentative refactoring of a handful of _mostly_ functionality tests based on decorated example functions

### CODE REVIEW

**Comment by jmlarson1** [comment]
@jlnav This greatly simplifies our many tests, which is nice. But I'm concerned it hides the necessary input/output relationships between the sims and gens. It is nice to have the decorators available, but we should think about how much we want to use them.

**Comment by jlnav** [comment]
> @jlnav This greatly simplifies our many tests, which is nice. But I'm concerned it hides the necessary input/output relationships between the sims and gens. It is nice to have the decorators available, but we should think about how much we want to use them.

Good point. We know for a given sim and gen that they're compatible if their outputs and inputs correspond. 

I just realized though, suppose a user tries to combine two user functions that are currently incompatible, their ins and outs don't match up. Changing the ins and outs within sim_specs and gen_specs _doesn't solve their incompatibility_! In fact, adjusting the ins/outs may cause the specs to pass validation, but errors would occur within the user function anyway.

I'll probably undo the regression test changes, since those are more user-examples.

File: libensemble/specs.py [hunk 1]
```diff
@@ -528,7 +530,9 @@ def persistent_uniform(_, persis_info, gen_specs, libE_info):
 
     def decorator(func):
         setattr(func, "persis_in", fields)
-        func.__doc__ = f"\n    **Persistent Input Fields:** ``{func.persis_inputs}``\n" + func.__doc__
+        if not func.__doc__:
+            func.__doc__ = ""
```
Comment (jmlarson1) [inline libensemble/specs.py:534]: Is there no test using `persis_in`? (Codecov says it's not covered)

File: libensemble/tests/scaling_tests/forces/forces_adv/run_libe_forces.py [hunk 2]
```diff
@@ -63,7 +63,6 @@
 # State the generating function, its arguments, output, and necessary parameters.
 gen_specs = {
     "gen_f": gen_f,  # Generator function
-    "in": [],  # Generator input
```
Comment (shuds13) [inline libensemble/tests/scaling_tests/forces/forces_adv/run_libe_forces.py:66]: Is removing this line necessary. I would rather not change it.

File: libensemble/tests/scaling_tests/forces/forces_simple/run_libe_forces.py [hunk 3]
```diff
@@ -41,7 +41,6 @@
 
     ensemble.gen_specs = GenSpecs(
         gen_f=gen_f,
-        inputs=[],  # No input when start persistent generator
```
Comment (shuds13) [inline libensemble/tests/scaling_tests/forces/forces_simple/run_libe_forces.py:44]: as above

File: libensemble/specs.py (hunk 1 above)
Comment (jlnav) [reply libensemble/specs.py:540]: These uncovered lines refer to user functions without docstrings, something that thankfully we have very few of! I'll see what I can do

**Review by shuds13 — APPROVED** [review]

**Pull request merged**